Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -5499,7 +5518,7 @@ template <class TreeBuilder> TreeExpression Parser<LexerType>::parseMemberExpres
 }
 
 template <typename LexerType>
-template <class TreeBuilder> TreeExpression Parser<LexerType>::parseArrowFunctionExpression(TreeBuilder& context, bool isAsync)
+template <class TreeBuilder> TreeExpression Parser<LexerType>::parseArrowFunctionExpression(TreeBuilder& context, std::optional<unsigned> functionStart)
```

Comment:
Let's have `isAsync` and `functionStart` separately. They have different meanings, and it is confusing that making `std::nullopt` functionStart as non-async.